src/message-io.ts

```diff
@@ -133,8 +135,26 @@ class MessageIO extends EventEmitter {
   /**
    * Read the next incoming message from the socket.
    */
-  async readMessage(): Promise<Message> {
-    const result = await this.incomingMessageIterator.next();
+  async readMessage(signal: AbortSignal): Promise<Message> {
+    let onAbort: () => void;
+    let onSocketError: (err: Error) => void;
+
+    const result = await Promise.race([
+      this.incomingMessageIterator.next(),
+
+      new Promise((resolve, reject) => {
+        onAbort = () => { reject(new AbortError()); };
+        signal.addEventListener('abort', onAbort);
+      }),
+
+      new Promise((resolve, reject) => {
+        onSocketError = reject;
+        this.socket.addListener('error', onSocketError);
+      })
```

Comment:
We have logic here to ensure that an `error` event while waiting for a new message is handled correctly. But I'm wondering what happens if a socket error happens before the call to `readMessage`? 🤔 and unhandled rejection maybe?